### PR TITLE
EventSetup consumes migration, SiStrip modules

### PR DESCRIPTION
**Pull request by wddgit**
#### PR description:

EventSetup consumes migration for SiStripMonitorTrack, and SiStripMonitorCluster. This required modifying the SiStripClusterInfo helper class.

Modified all the other clients of SiStripClusterInfo also to keep working with the changes in the helper class, but I didn't finish the migration for those other modules because I am not sure which are actually still used. But the part using SiStripClusterInfo is migrated now.

Note that I deleted the reclusterize function from SiStripClusterInfo. Nothing in the repository used that function and it seemed an odd place to run clusterization from anyway. I deleted it because the reclusterize function used EventSetup and removing that postpones the EventSetup consumes migration of that algorithm until we migrate the module that actually runs it.

#### PR validation:

Relies on existing tests. There is no new functionality. All results should be identical if I've done this correctly.

### CODE REVIEW

**Comment by wddgit** [comment]
please test

**Comment by jfernan2** [comment]
+dqm

**Comment by slava77** [comment]
> Note that I deleted the reclusterize function from SiStripClusterInfo.

@pieterdavid @alesaggio 
please check/confirm that this method is currently not used for anything of importance

**Comment by pieterdavid** [comment]
@slava77 indeed, this method isn't used anywhere that I know of (it could be useful for studies of clusterizer algorithms and settings, but it can easily be revived for that if/when needed, without complicating this code)

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/28683	41c7cce 
- technical refactoring, in line with the PR description. The removal of an unused method was confirmed OK.
- jenkins tests pass and comparisons with the baseline show no (relevant) differences

File: CalibTracker/SiStripHitEfficiency/src/HitEff.cc [hunk 1]
```diff
@@ -683,8 +685,8 @@ void HitEff::analyze(const edm::Event& e, const edm::EventSetup& es) {
                     cluster_info.push_back(sqrt(parameters.second.xx()));
                     cluster_info.push_back(parameters.first.y());
                     cluster_info.push_back(sqrt(parameters.second.yy()));
-                    cluster_info.push_back(clusterInfo.signalOverNoise());
-                    //cluster_info.push_back( clusterInfo.getSignalOverNoise() );
+                    cluster_info.push_back(siStripClusterInfo_.signalOverNoise());
+                    //cluster_info.push_back( siStripClusterInfo_.getSignalOverNoise() );
```
Comment (christopheralanwest) [inline CalibTracker/SiStripHitEfficiency/src/HitEff.cc]: `getSignalOverNoise()` appears nowhere else in CMSSW. Can we simply remove the commented line?

Comment (wddgit) [reply CalibTracker/SiStripHitEfficiency/src/HitEff.cc]: Done. That is better, thanks.

**Comment by wddgit** [comment]
Please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/28683	e8efbb2 
- changes since the last reco signoff are unrelated to reco (commented line removed) https://github.com/cms-sw/cmssw/pull/28683/files/41c7cce6776d9250d11d1d1b799e281e64ee628a..e8efbb29f49a0100d797295123ab8ab62da29f5c

**Comment by jfernan2** [comment]
+dqm

**Comment by santocch** [comment]
+1

**Comment by silviodonato** [comment]
@christopheralanwest  @franzoni  @tlampen  @pohsun  @tocheng please review/sign

**Comment by tlampen** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**